boards/arm64/qemu_kvm_arm64/board.cmake

```diff
@@ -24,6 +24,12 @@ if(CONFIG_XIP)
   set(QEMU_KERNEL_OPTION
   -bios ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
   )
+  board_runner_args(qemu "--commander=qemu-system-aarch64" "--cpu=host"
+    "--machine=${QEMU_MACH}"
+    "--qemu_option=${CMAKE_CURRENT_LIST_DIR}/qemu_kvm_arm64_xip_option.yaml")
```

Comment:
this is a general comment which applies to most of the `board_runner_args`.

Are there any reasons why we must maintain the same arguments here: https://github.com/zephyrproject-rtos/zephyr/blob/9d6f192197d7e122edd8fd5204d1c751eb5aa7ee/boards/arm64/qemu_kvm_arm64/board.cmake#L14-L17
and in the lines where my comment is placed ?

We should not have two locations for specifying the same flags.
I think it makes a lot of sense to move the qemu to `west`, but then I believe the CMake target should be reworked into simply invoking `west` like it's done with `flash`.

Note: would like input from people not using `west`, as to how such a change might impact them.
FYI @daor-oti @mped-oticon

Reply:
the west can not get all the parameters set in CMake. and I do not want to keep adding paramenters to the board_runner_args. so I put those static options and west comand obtainable options to the qemu_xxx_option.yaml. and put the cmake varibale options which did not pass to west in to a qemu_runner_variables.conf file, which I used only in qemu_x86, as too many CMake run time options added.

this PR is to make west to support multiply simulator runner (for now only qemu is added), which cmakefile can't support well. But I keep the cmake system unchanged,  and can be phase out.

Reply:
> the west can not get all the parameters set in CMake.

i'm well aware of that, and hence the reason for this comment:
`but then I believe the CMake target should be reworked into simply invoking west like it's done with flash.`

We don't want to maintain two implementations that tries to fulfill the same.
In this case, one implementation (the CMake) should be reworked into a wrapper of the real and maintained implementation.